### PR TITLE
dev, 3.0: Update aws deployment document

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

### What is changed, added or deleted? 
Update aws deployment document

### What is the related PR or file link(s)? 
https://github.com/pingcap/tidb-operator/pull/650

### Which version does your change affect? 
`dev` `3.0`

### Checklist 

- [x] Add a new file to `TOC.md`
- [x] No Tab spaces anywhere 
- [x] Leave a blank line both before and after a code block
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

File: dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md [hunk 1]
```diff
@@ -247,10 +247,81 @@ TiDB 版本和组件数量也可以在 `variables.tf` 中修改，你可以按
...
+  # The version of tidb-cluster helm chart
+  tidb_cluster_chart_version    = "v1.0.0-rc.1"
+}
+```
+
+> **注意：**
+>
+> `cluster_name` 必须是唯一的
+
+你可以通过 `kubectl` 获取新集群的监控系统地址与 TiDB 地址，假如你希望让 Terraform 脚本输出这些地址，可以通过在 `outputs.tf` 中增加相关的输出项实现：
```
Comment (dcalvin) [inline dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]:
```suggestion
你可以通过 `kubectl` 获取新集群的监控系统地址与 TiDB 地址。假如你希望让 Terraform 脚本输出这些地址，可以通过在 `outputs.tf` 中增加相关的输出项实现：
```

File: dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md [hunk 2]
```diff
@@ -262,6 +333,154 @@ TiDB 版本和组件数量也可以在 `variables.tf` 中修改，你可以按
 terraform destroy
 ```
 
+> **注意：**
+>
+> 该操作会销毁 EKS 集群以及部署在该 EKS 集群上的所有 TiDB 集群。
+
 > **注意：**
```
Comment (dcalvin) [inline dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]: I think we can combine this two notes into one.

File: dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md [hunk 3]
```diff
@@ -262,6 +333,154 @@ TiDB 版本和组件数量也可以在 `variables.tf` 中修改，你可以按
...
+
+## 管理多个 Kubernetes 集群
+
+本节详细介绍了如何管理多个 Kubernetes 集群（EKS），并在每个集群上部署一个或更多 TiDB 集群。
+
+上述文档中介绍的 Terraform 脚本组合了多个 Terraform 模块：
+
+- `tidb-operator` 模块，用于创建 EKS 集群并在 EKS 集群上安装配置 [TiDB Operator](how-to/deploy/tidb-operator.md)。
+- `tidb-cluster` 模块，用于创建 TiDB 集群所需的资源池并部署 TiDB 集群。
+- 以及 EKS 上的 TiDB 集群专用的 `vpc` 模块、`key-pair`模块和`bastion` 模块
```
Comment (dcalvin) [inline dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]:
```suggestion
- EKS 上的 TiDB 集群专用的 `vpc` 模块、`key-pair`模块和`bastion` 模块
```

File: dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md [hunk 4]
```diff
@@ -262,6 +333,154 @@ TiDB 版本和组件数量也可以在 `variables.tf` 中修改，你可以按
...
+
+上面的例子很容易进行定制，比如，假如你不需要堡垒机，便可以删去对 `bastion` 模块的调用。同时，项目中提供的 Terraform 模块均设置了合理的默认值，因此在调用这些 Terraform 模块时，你可以略去大部分的参数。
+
+你可以参考默认的 Terraform 脚本来定制每个模块的参数，也可以参考每个模块的 `variables.tf` 文件来了解所有可配置的参数。
+
+另外，这些 Terraform 模块可以很容易地集成到你自己的 Terraform 工作流中。假如你对 Terraform 非常熟悉，这也是我们推荐的一种使用方式
+
+> **注意：**
+>
+> 由于 Terraform 本身的限制（[hashicorp/terraform#2430](https://github.com/hashicorp/terraform/issues/2430#issuecomment-370685911)），在你自己的 Terraform 脚本中，也需要保留上述例子中对 `helm provider` 的特殊处理
```
Comment (dcalvin) [inline dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]:
```suggestion
> 由于 Terraform 本身的限制（[hashicorp/terraform#2430](https://github.com/hashicorp/terraform/issues/2430#issuecomment-370685911)），在你自己的 Terraform 脚本中，也需要保留上述例子中对 `helm provider` 的特殊处理。
```

File: dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md [hunk 5]
```diff
@@ -262,6 +333,154 @@ TiDB 版本和组件数量也可以在 `variables.tf` 中修改，你可以按
...
+
+> **注意：**
+>
+> 创建新目录时，需要注意与 Terraform 模块之间的相对路径，这会影响调用模块时的 `source` 参数。
+
+> **注意：**
+>
+> 假如你想在 tidb-operator 项目之外使用这些模块，你需要确保 `modules` 目录中的所有模块的相对路径保持不变。
```
Comment (dcalvin) [inline dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]: Combine these Notes into one using unordered list.

**Comment by aylei** [comment]
@dcalvin Thanks for your careful review! I've addressed all the comments, could you please take a look again?

File: dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md [hunk 6]
```diff
@@ -113,15 +114,14 @@ Apply complete! Resources: 67 added，0 changed，0 destroyed.
 Outputs:
 
 bastion_ip = [
-    52.14.50.145
+  "34.219.204.217",
 ]
-eks_endpoint = https://E10A1D0368FFD6E1E32E11573E5CE619.sk1.us-east-2.eks.amazonaws.com
+default-cluster_monitor-dns = a82db513ba84511e9af170283460e413-1838961480.us-west-2.elb.amazonaws.com
```
Comment (DanielZhangQD) [inline dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]: Suggest to print ports info.

File: dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md [hunk 7]
```diff
@@ -264,4 +335,143 @@ terraform destroy
...
+  description = "Bastion IP address"
+  value       = module.bastion.bastion_ip
+}
+```
+
+上面的例子很容易进行定制，比如，假如你不需要堡垒机，便可以删去对 `bastion` 模块的调用。同时，项目中提供的 Terraform 模块均设置了合理的默认值，因此在调用这些 Terraform 模块时，你可以略去大部分的参数。
+
+你可以参考默认的 Terraform 脚本来定制每个模块的参数，也可以参考每个模块的 `variables.tf` 文件来了解所有可配置的参数。
+
+另外，这些 Terraform 模块可以很容易地集成到你自己的 Terraform 工作流中。假如你对 Terraform 非常熟悉，这也是我们推荐的一种使用方式
```
Comment (DanielZhangQD) [inline dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]: Miss 。 in the end.

File: dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md (hunk 6 above)
Comment (aylei) [reply dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]: It's an enhancement which has not yet been made

File: dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md (hunk 7 above)
Comment (aylei) [inline dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]:
```suggestion
另外，这些 Terraform 模块可以很容易地集成到你自己的 Terraform 工作流中。假如你对 Terraform 非常熟悉，这也是我们推荐的一种使用方式。
```

File: v3.0/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md [hunk 8]
```diff
@@ -265,4 +335,143 @@ terraform destroy
...
+  description = "Bastion IP address"
+  value       = module.bastion.bastion_ip
+}
+```
+
+上面的例子很容易进行定制，比如，假如你不需要堡垒机，便可以删去对 `bastion` 模块的调用。同时，项目中提供的 Terraform 模块均设置了合理的默认值，因此在调用这些 Terraform 模块时，你可以略去大部分的参数。
+
+你可以参考默认的 Terraform 脚本来定制每个模块的参数，也可以参考每个模块的 `variables.tf` 文件来了解所有可配置的参数。
+
+另外，这些 Terraform 模块可以很容易地集成到你自己的 Terraform 工作流中。假如你对 Terraform 非常熟悉，这也是我们推荐的一种使用方式
```
Comment (aylei) [inline v3.0/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]:
```suggestion
另外，这些 Terraform 模块可以很容易地集成到你自己的 Terraform 工作流中。假如你对 Terraform 非常熟悉，这也是我们推荐的一种使用方式。
```

**Comment by aylei** [comment]
I've addressed all the review comments, PTAL again @dcalvin @DanielZhangQD

File: dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md [hunk 9]
```diff
@@ -237,7 +239,8 @@ Grafana 默认登录信息：
 
 > **注意：**
 >
-> 由于 AWS 和 Terraform 的限制，还不支持复用已有 EKS 集群的 VPC 和 subnets，所以请确保只在你手动创建 VPC 的情况下修改该参数。
+> - 由于 AWS 和 Terraform 的限制，还不支持复用已有 EKS 集群的 VPC 和 subnets，所以请确保只在你手动创建 VPC 的情况下修改该参数；
+> - EKS Node 上的 CNI 插件会为每个节点预留一部分 IP 资源，因此 IP 消耗较大，在手动创建 VPC 时，建议将每个 subnet 的掩码长度设置在 18~20 以确保 IP 资源充足；
```
Comment (tennix) [inline dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]: Add link for the CNI, users can also reduce the default warmup IP for a node.

File: dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md [hunk 10]
```diff
@@ -237,7 +239,8 @@ Grafana 默认登录信息：
...
 
 由于 TiDB 服务通过 [Internal Elastic Load Balancer](https://aws.amazon.com/blogs/aws/internal-elastic-load-balancers/) 暴露，默认情况下，会创建一个 ec2 实例作为堡垒机，访问创建的 TiDB 集群。堡垒机上预装了 MySQL 和 Sysbench，所以你可以 SSH 到堡垒机然后通过 ELB 访问 TiDB。如果你的 VPC 中已经有了类似的 ec2 实例，你可以通过设置 `create_bastion` 为 `false` 禁掉堡垒机的创建。
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]: SSH 不是动词。。

[the remark this answers is not in the document]
Comment (aylei) [reply dev/how-to/deploy/orchestrated/tidb-in-kubernetes/aws-eks.md]: 开发平时这么说惯了，写岔了😂

**Comment by aylei** [comment]
@tennix @dcalvin PTAL again

**Comment by aylei** [comment]
@dcalvin Can we merge this?

**Pull request merged**